baal/active/heuristics/heuristics.py

```diff
@@ -725,3 +725,146 @@ def reorder_indices(self, scores_list):
             ranks = ranks[::-1]
         ranks = _shuffle_subset(ranks, self.shuffle_prop)
         return ranks
+
+
+class EPIG(AbstractHeuristic):
+    """
+    Implementation of Expected Predicted Information Gain
+    https://arxiv.org/abs/2304.08151
+
+    References:
+    Code from https://github.com/fbickfordsmith/epig
+    """
+
+    def __init__(self, shuffle_prop=DEPRECATED, reverse=False, reduction="none"):
+        super().__init__(shuffle_prop=shuffle_prop, reverse=True, reduction=reduction)
+
+    def entropy_from_probs(self,probs):
+        """
+        See entropy_from_logprobs.
+
+        If p(y=y'|x) is 0, we make sure p(y=y'|x) log p(y=y'|x) evaluates to 0, not NaN.
+
+        Arguments:
+            probs: Tensor[float], [*N, Cl]
+
+        Returns:
+            Tensor[float], [*N,]
+        """
+        logprobs = torch.clone(probs)  #  [*N, Cl]
+        logprobs[probs > 0] = torch.log(probs[probs > 0])  #  [*N, Cl]
+        return -torch.sum((probs * logprobs), dim=-1)  # [*N,]
```

Comment:
This can all be cut to `return -torch.sum(torch.xlogy(probs, probs), dim=-1)`

Reply:
Very nice!